notification-support/src/main/java/com/expediagroup/streamplatform/streamregistry/core/events/KafkaSetupHandler.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (C) 2018-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.streamplatform.streamregistry.core.events;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.CreateTopicsOptions;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.errors.UnknownTopicOrPartitionException;
+
+@Slf4j
+@Builder
+public class KafkaSetupHandler {
+    public static final int DEFAULT_CREATE_TOPIC_TIMEOUT_MS = 10000;
+
+    @NonNull
+    @Getter
+    private final String bootstrapServers;
+
+    @NonNull
+    @Getter
+    private final Optional<NewTopic> newTopic;
```

Comment:
Should this be `Optional`? The only use of it assumes it's present by doing `get()` so doesn't seem to add any value. The conditional seem to lie in `isKafkaSetupEnabled`.

Reply:
Fields should _never_ be Optional.

Reply:
The Java language authors have been quite clear that Optional was intended for use only as a return type, as a way to convey that a method may or may not return a value.

Using Optional requires us to reason with three cases (null, present, absent) rather than 2 (null, present), introduces a new Object, and buy us anything. 

Having a @NonNull Optional<T> doesn't improve on having a nullable T

Reply:
I see that this has the @getter annotation, but I still don't like it.

Reply:
I decided to use Optional in new topic cause the creation of depends on a condition, but a boolean should be clearer than it, I think a over use it here :).
My reason to use lombok's `@NotNull` is because bean is not build automagically by spring context but in the config class and it will throw an error if isn't defined, but you are absolutely right, it should be solved in the constructor, builder or with a boolean in a meaningful way. (added to my modification list)

Reply:
Solved